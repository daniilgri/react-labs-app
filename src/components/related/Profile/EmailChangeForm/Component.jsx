import React from "react";
import { useFormik } from "formik";
import PropTypes from "prop-types";

import emailChangeSchema from "../../../../validations/emailChangeSchema";
import { Wrapper, Head, Title, Body, Field, Label, Input, ErrorText, FilledButton } from "./styles";

const Component = ({ changeEmailRequested }) => {
  const formik = useFormik({
    initialValues: {
      emailChange: "",
      passwordConfirmEmail: "",
    },
    validationSchema: emailChangeSchema,
    onSubmit: (values, { resetForm }) => {
      changeEmailRequested({
        email: values.emailChange,
        password: values.passwordConfirmEmail,
      });
      resetForm();
    },
  });

  return (
    <Wrapper onSubmit={formik.handleSubmit} autoComplete="off">
      <Head>
        <Title>Email change</Title>
      </Head>
      <Body>
        <Field>
          <Label htmlFor="emailChange">Email</Label>
          <Input
            id="emailChange"
            name="emailChange"
            placeholder="New email"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.emailChange}
            autocomplete="nope"
          />
          {formik.errors.emailChange && <ErrorText>{formik.errors.emailChange}</ErrorText>}
        </Field>
        <Field>
          <Label htmlFor="passwordConfirmEmail">Password to confirm email change</Label>
          <Input
            id="passwordConfirmEmail"
            name="passwordConfirmEmail"
            placeholder="Current password"
            type="password"
            onChange={formik.handleChange}
            value={formik.values.passwordConfirmEmail}
            autocomplete="new-password"
          />
          {formik.errors.passwordConfirmEmail && (
            <ErrorText>{formik.errors.passwordConfirmEmail}</ErrorText>
          )}
        </Field>
      </Body>
      <FilledButton type="submit">Change</FilledButton>
    </Wrapper>
  );
};

Component.propTypes = {
  changeEmailRequested: PropTypes.func.isRequired,
};

export default Component;
