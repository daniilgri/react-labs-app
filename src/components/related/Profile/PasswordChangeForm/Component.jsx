import React from "react";
import { useFormik } from "formik";
import PropTypes from "prop-types";

import changePasswordSchema from "../../../../validations/changePasswordSchema";
import {
  Wrapper,
  Head,
  Title,
  Body,
  Field,
  Label,
  Input,
  ErrorText,
  FilledButton,
} from "./styles";

const Component = ({ changePasswordRequested }) => {
  const formik = useFormik({
    initialValues: {
      currentPassword: "",
      newPassword: "",
      confirmPasswordChange: "",
    },
    validationSchema: changePasswordSchema,
    onSubmit: (values) => {
      changePasswordRequested({
        password: values.currentPassword,
        newPassword: values.newPassword,
      });
    },
  });

  return (
    <Wrapper onSubmit={formik.handleSubmit}>
      <Head>
        <Title>Password change</Title>
      </Head>
      <Body>
        <Field>
          <Label htmlFor="currentPassword">Current password</Label>
          <Input
            id="currentPassword"
            name="currentPassword"
            placeholder="Current password"
            type="password"
            onChange={formik.handleChange}
            value={formik.values.currentPassword}
          />
          {formik.errors.currentPassword && (
            <ErrorText>{formik.errors.currentPassword}</ErrorText>
          )}
        </Field>
        <Field>
          <Label htmlFor="newPassword">New password</Label>
          <Input
            id="newPassword"
            name="newPassword"
            placeholder="New password"
            type="password"
            onChange={formik.handleChange}
            value={formik.values.newPassword}
          />
          {formik.errors.newPassword && (
            <ErrorText>{formik.errors.newPassword}</ErrorText>
          )}
        </Field>
        <Field>
          <Label htmlFor="confirmPasswordChange">Confirm password</Label>
          <Input
            id="confirmPasswordChange"
            name="confirmPasswordChange"
            placeholder="Confirm password"
            type="password"
            onChange={formik.handleChange}
            value={formik.values.confirmPasswordChange}
          />
          {formik.errors.confirmPasswordChange && (
            <ErrorText>{formik.errors.confirmPasswordChange}</ErrorText>
          )}
        </Field>
      </Body>
      <FilledButton>Change</FilledButton>
    </Wrapper>
  );
};

Component.defaultProps = {
  changePasswordRequested: () => {},
};

Component.propTypes = {
  changePasswordRequested: PropTypes.func.isRequired,
};

export default Component;
