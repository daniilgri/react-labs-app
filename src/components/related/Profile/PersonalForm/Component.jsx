import React from "react";
import PropTypes from "prop-types";
import { useFormik } from "formik";

import personalInfoSchema from "../../../../validations/personalInfoSchema";
import {
  Wrapper,
  Head,
  Title,
  Body,
  Field,
  Label,
  Input,
  ErrorText,
  FilledButton,
  Loading,
} from "./styles";

const Component = ({ user, loading, error, updateProfileRequested }) => {
  const formik = useFormik({
    initialValues: {
      firstNameChange: user.firstName,
      lastNameChange: user.lastName,
    },
    enableReinitialize: true,
    validationSchema: personalInfoSchema,
    onSubmit: values => {
      const changedValues = {};

      Object.entries(values).forEach((value, key) => {
        if (formik.initialValues[key] !== value) {
          changedValues[key] = value;
        }
      });
      if (Object.keys(changedValues).length !== 0 && changedValues.constructor === Object) {
        updateProfileRequested(changedValues);
      }
    },
  });

  if (loading || error) {
    return <Loading>Loading...</Loading>;
  }

  return (
    <Wrapper onSubmit={formik.handleSubmit}>
      <Head>
        <Title>Personal information</Title>
      </Head>
      <Body>
        <Field>
          <Label htmlFor="firstNameChange">First name</Label>
          <Input
            id="firstNameChange"
            name="firstNameChange"
            placeholder="First name"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.firstNameChange}
          />
          {formik.errors.firstNameChange && <ErrorText>{formik.errors.firstNameChange}</ErrorText>}
        </Field>
        <Field>
          <Label htmlFor="lastNameChange">Last name</Label>
          <Input
            id="lastNameChange"
            name="lastNameChange"
            placeholder="Last name"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.lastNameChange}
          />
          {formik.errors.lastNameChange && <ErrorText>{formik.errors.lastNameChange}</ErrorText>}
        </Field>
      </Body>
      <FilledButton type="submit">Save</FilledButton>
    </Wrapper>
  );
};

Component.defaultProps = {
  error: "",
};

Component.propTypes = {
  user: PropTypes.objectOf(PropTypes.object.isRequired).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  updateProfileRequested: PropTypes.func.isRequired,
};

export default Component;
