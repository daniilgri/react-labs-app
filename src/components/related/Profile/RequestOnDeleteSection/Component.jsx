import React from "react";
import PropTypes from "prop-types";
import { useFormik } from "formik";

import { Wrapper, Label, FilledButton, Form, Input, Bold } from "./styles";

const Component = ({
  user,
  loading,
  error,
  requestOnDeleteRequested,
  cancelRequestOnDeleteRequested,
}) => {
  const formik = useFormik({
    initialValues: {
      requestOnDeleteEmail: "",
    },
    onSubmit: (values, { resetForm }) => {
      if (user.email === values.requestOnDeleteEmail) {
        requestOnDeleteRequested({ uid: user.uid });
        resetForm();
      }
    },
  });

  const handleCancelButtonOnClick = event => {
    event.preventDefault();
    cancelRequestOnDeleteRequested({ uid: user.uid });
  };

  if (loading || error) {
    return null;
  }

  return (
    <Wrapper>
      <Label>
        Wanna delete your account? Enter your e-mail address
        <Bold>{user.email}</Bold>
        to confirm
      </Label>
      <Form onSubmit={formik.handleSubmit}>
        <Input
          id="requestOnDeleteEmail"
          name="requestOnDeleteEmail"
          onChange={formik.handleChange}
          value={formik.values.requestOnDeleteEmail}
          disabled={user.requestOnDelete}
        />
        <FilledButton bgColor="#f85d4b" disabled={user.requestOnDelete} type="submit">
          Confirm
        </FilledButton>
        {user.requestOnDelete && (
          <FilledButton bgColor="#e24545" type="button" onClick={handleCancelButtonOnClick}>
            Cancel
          </FilledButton>
        )}
      </Form>
    </Wrapper>
  );
};

Component.defaultProps = {
  user: null,
  error: "",
};

Component.propTypes = {
  user: PropTypes.objectOf(PropTypes.object),
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  requestOnDeleteRequested: PropTypes.func.isRequired,
  cancelRequestOnDeleteRequested: PropTypes.func.isRequired,
};

export default Component;
