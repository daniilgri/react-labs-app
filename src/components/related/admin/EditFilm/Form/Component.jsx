import React, { useEffect } from "react";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";
import { useFormik } from "formik";

import DescriptionSection from "./DescriptionSection";
import MediaSection from "./MediaSection";
import FormalSection from "./FormalSection";

import { Wrapper, Head, Title, Body, FilledButton, Loading, RedirectLink } from "./styles";

const Component = ({
  openAddScreeningDateModal,
  filmId,
  film,
  editFilmRequested,
  fetchEditFilmRequested,
  loading,
  error,
}) => {
  useEffect(() => {
    fetchEditFilmRequested(filmId);
  }, [fetchEditFilmRequested, filmId]);

  const formik = useFormik({
    initialValues: {
      title: film.title,
      description: film.description,
      ticketPrice: film.ticketPrice,
      imageAsFile: film.image,
      previewImage: film.image,
      tags: film.tags,
      screeningDates: film.screeningDates,
    },
    onSubmit: values => {
      const changedValues = {};
      Object.keys(values).forEach(prop => {
        if (formik.initialValues[prop] !== values[prop]) {
          changedValues[prop] = values[prop];
        }
      });
      if (Object.keys(changedValues).length !== 0 && changedValues.constructor === Object) {
        editFilmRequested({ values, changedValues, filmId: film.id });
      }
    },
    enableReinitialize: true,
  });

  const handleImageUpload = event => {
    event.preventDefault();
    formik.setFieldValue("imageAsFile", event.currentTarget.files[0]);
  };

  const handleScreeningDateAddition = screeningDate => {
    formik.setFieldValue("screeningDates", [...formik.values.screeningDates, screeningDate]);
  };

  const handleScreeningDateDelete = screeningDate => {
    formik.setFieldValue(
      "screeningDates",
      formik.values.screeningDates.filter(sd => sd.date !== screeningDate.date)
    );
  };

  const handleTagAddition = tag => {
    formik.setFieldValue("tags", [...formik.values.tags, tag]);
  };

  const handleTagDelete = tag => {
    formik.setFieldValue(
      "tags",
      formik.values.tags.filter(elTag => elTag !== tag)
    );
  };

  const handleFormOnKeyDown = event => {
    if (event.key === "Enter") {
      event.preventDefault();
    }
  };

  if (loading || error) {
    return <Loading>Loading...</Loading>;
  }

  return (
    <Wrapper onKeyDown={handleFormOnKeyDown} onSubmit={formik.handleSubmit}>
      <Head>
        <Title>Edit film - {film.title}</Title>{" "}
        <RedirectLink to={`/admin/film/${film.id}`}>Go to film card</RedirectLink>
      </Head>
      <Body>
        <DescriptionSection
          title={formik.values.title}
          description={formik.values.description}
          tags={formik.values.tags}
          errors={formik.errors}
          onChange={formik.handleChange}
          onTagAdd={handleTagAddition}
          onTagDelete={handleTagDelete}
        />
        <MediaSection
          errors={formik.errors}
          onImageUpload={handleImageUpload}
          previewImage={formik.values.previewImage}
        />
        <FormalSection
          screeningDates={formik.values.screeningDates}
          ticketPrice={formik.values.ticketPrice}
          errors={formik.errors}
          onChange={formik.handleChange}
          onAddScreeningDateModalOpen={openAddScreeningDateModal}
          onScreeningDateAdd={handleScreeningDateAddition}
          onScreeningDateDelete={handleScreeningDateDelete}
        />
      </Body>

      <FilledButton type="submit">Publish</FilledButton>
    </Wrapper>
  );
};

Component.defaultProps = {
  error: "",
  filmId: "",
};

Component.propTypes = {
  editFilmRequested: PropTypes.func.isRequired,
  openAddScreeningDateModal: PropTypes.func.isRequired,
  filmId: PropTypes.string,
  film: PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.bool, PropTypes.number, PropTypes.string, PropTypes.array])
  ).isRequired,
  fetchEditFilmRequested: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
};

export default withRouter(Component);
