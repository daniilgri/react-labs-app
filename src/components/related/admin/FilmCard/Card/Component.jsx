import React, { useEffect } from "react";
import PropTypes from "prop-types";

import FilmBox from "./FilmBox";
import TicketBox from "./TicketBox";
import { Wrapper, Pending } from "./styles";

const Component = ({ filmId, fetchFilmByIdAdminPanelRequested, film, filmError, loading }) => {
  useEffect(() => {
    fetchFilmByIdAdminPanelRequested(filmId);
  }, [fetchFilmByIdAdminPanelRequested, filmId]);

  if (loading || filmError || !film) {
    return <Pending>Loading</Pending>;
  }

  return (
    <Wrapper>
      <FilmBox film={film} />
      <TicketBox screeningDates={film.screeningDates} />
    </Wrapper>
  );
};

Component.defaultProps = {
  filmError: "",
  filmId: "",
  film: {
    title: "",
    description: "",
    tags: [],
    image: "",
    rates: [],
    screeningDates: [],
    ticketPrice: "",
  },
};

Component.propTypes = {
  filmId: PropTypes.string,
  fetchFilmByIdAdminPanelRequested: PropTypes.func.isRequired,
  film: PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.bool, PropTypes.number, PropTypes.string, PropTypes.array])
  ),
  filmError: PropTypes.string,
  loading: PropTypes.bool.isRequired,
};

export default Component;
