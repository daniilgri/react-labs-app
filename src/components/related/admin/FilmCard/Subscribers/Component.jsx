import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { v4 as uuidv4 } from "uuid";
import Card from "./Card";

import { Wrapper, Loading, CenterContainer, FetchButton } from "./styles";

const Component = ({
  filmId,
  fetchSubscribersInitialRequested,
  users,
  loading,
  error,
  allCount,
  count,
  fetchSubscribersNextRequested,
}) => {
  const handleGetMoreButtonOnClick = event => {
    event.preventDefault();

    if (count < allCount) {
      fetchSubscribersNextRequested({ filmId });
    }
  };

  useEffect(() => {
    fetchSubscribersInitialRequested({ filmId });
  }, []);

  return (
    <>
      <Wrapper>
        {users.length > 0 && users.map(item => <Card key={uuidv4()} user={item} />)}
      </Wrapper>
      {loading || error ? (
        <Loading>Loading</Loading>
      ) : (
        count < allCount && (
          <CenterContainer>
            <FetchButton onClick={handleGetMoreButtonOnClick}>Get more</FetchButton>
          </CenterContainer>
        )
      )}
    </>
  );
};

Component.defaultProps = {
  users: [],
  error: "",
  count: 0,
  allCount: 0,
  filmId: "",
};

Component.propTypes = {
  filmId: PropTypes.string,
  fetchSubscribersInitialRequested: PropTypes.func.isRequired,
  users: PropTypes.arrayOf(PropTypes.object),
  error: PropTypes.string,
  loading: PropTypes.bool.isRequired,
  count: PropTypes.number,
  allCount: PropTypes.number,
  fetchSubscribersNextRequested: PropTypes.func.isRequired,
};

export default Component;
