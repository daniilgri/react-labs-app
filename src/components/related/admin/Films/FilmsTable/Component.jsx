import React, { useEffect } from "react";
import PropTypes from "prop-types";
import Card from "./Card";
import { v4 as uuidv4 } from "uuid";

import { Wrapper, Loading } from "./styles";

const Component = ({
  loading,
  fetchFilmsAdminPanelInitialRequested,
  films,
  error,
  deleteFilmRequested,
}) => {
  useEffect(() => {
    fetchFilmsAdminPanelInitialRequested({ count: 25 });
  }, []);

  let content;

  if (loading || error) {
    content = <Loading>Loading</Loading>;
  } else {
    content =
      films.length > 0 ? (
        films.map((item) => (
          <Card key={uuidv4()} film={item} onDelete={deleteFilmRequested} />
        ))
      ) : (
        <Loading>Empty</Loading>
      );
  }

  return <Wrapper>{content}</Wrapper>;
};

Component.defaultProps = {
  films: [],
  loading: true,
  error: "",
  fetchFilmsAdminPanelInitialRequested: () => {},
  deleteFilmRequested: () => {},
};

Component.propTypes = {
  films: PropTypes.array,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  fetchFilmsAdminPanelInitialRequested: PropTypes.func.isRequired,
  deleteFilmRequested: PropTypes.func.isRequired,
};

export default Component;
