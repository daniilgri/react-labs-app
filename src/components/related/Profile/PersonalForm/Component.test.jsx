import React from "react";
import renderer from "react-test-renderer";

import Component from "./Component";

describe("PersonalForm component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Component
          updateProfileRequested={() => {}}
          user={{
            firstName: "firstName",
            lastName: "lastName",
            uid: "e12",
            email: "fkfk@gmail.com",
            role: "guest",
            requestOnDelete: false,
          }}
          loading={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
