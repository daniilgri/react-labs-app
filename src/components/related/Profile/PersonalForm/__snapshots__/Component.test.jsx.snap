// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalForm component renders correctly 1`] = `
<form
  className="sc-bdfBwQ fRzuAI"
  onSubmit={[Function]}
>
  <div
    className="sc-gsTCUz koJJD"
  >
    <h2
      className="sc-dlfnbm iLCklO"
    >
      Personal information
    </h2>
  </div>
  <div
    className="sc-hKgILt eCHxNa"
  >
    <div
      className="sc-eCssSg kbddLr"
    >
      <label
        className="sc-jSgupP hhWncP"
        htmlFor="firstNameChange"
      >
        First name
      </label>
      <input
        className="sc-gKsewC hHOZFO"
        id="firstNameChange"
        name="firstName"
        onChange={[Function]}
        placeholder="First name"
        type="text"
        value="firstName"
      />
    </div>
    <div
      className="sc-eCssSg kbddLr"
    >
      <label
        className="sc-jSgupP hhWncP"
        htmlFor="lastNameChange"
      >
        Last name
      </label>
      <input
        className="sc-gKsewC hHOZFO"
        id="lastNameChange"
        name="lastName"
        onChange={[Function]}
        placeholder="Last name"
        type="text"
        value="lastName"
      />
    </div>
  </div>
  <button
    className="sc-fubCfw hTaewY"
    type="submit"
  >
    Save
  </button>
</form>
`;
